solutions/2200-2299/2229-check-if-an-array-is-consecutive-easy.md

```diff
@@ -0,0 +1,89 @@
+---
+description: >-
+  Author: @hiverkiya | https://leetcode.com/problems/check-if-an-array-is-consecutive/
+  tags: [Array]
+---
+
+# 2229 - Check if an Array is Consecutive (Easy)
+
+## Problem Link
+
+https://leetcode.com/problems/check-if-an-array-is-consecutive/
+
+## Problem Statement
+
+Given an integer array `nums`, return `true` _if_ `nums` _is_ **_consecutive_** _, otherwise return_ `false`.
+
+An array is **consecutive** if it contains every number in the range `[x, x + n - 1]` **(inclusive)**, where `x` is the minimum number in the array and `n` is the length of the array.
+
+**Example 1:**
+
+```
+Input: nums = [1,3,4,2]
+Output: true
+Explanation:
+The minimum value is 1 and the length of nums is 4.
+All of the values in the range [x, x + n - 1] = [1, 1 + 4 - 1] = [1, 4] = (1, 2, 3, 4) occur in nums.
+Therefore, nums is consecutive.
+```
+
+**Example 2:**
+
+```
+Input: nums = [1,3]
+Output: false
+Explanation:
+The minimum value is 1 and the length of nums is 2.
+The value 2 in the range [x, x + n - 1] = [1, 1 + 2 - 1], = [1, 2] = (1, 2) does not occur in nums.
+Therefore, nums is not consecutive.
+```
+**Example 3:**
+```
+Input: nums = [3,5,4]
+Output: true
+Explanation:
+The minimum value is 3 and the length of nums is 3.
+All of the values in the range [x, x + n - 1] = [3, 3 + 3 - 1] = [3, 5] = (3, 4, 5) occur in nums.
+Therefore, nums is consecutive.
+```
+
+**Constraints:**
+
+- `1 <= nums.length <= 105`
+- `0 <= nums[i] <= 105`
+
+## Approach 1: Sort and Compare
+
+The problem statements require you to check for consecutive elements in a range, which means the element next to the current element should be one number higher. For example, [1,2,3] are consecutive elements while [1,3,4] or [1,2,2,3,4] are not. That means the **current element + 1 = next element**.
+
+The elements are given in random order, and it's mentioned that the elements are in a range [x,x+n-1] with x as the minimum element. To find the minimum element, you can sort the array in ascending order, and the first element will be the minimum element.
+
+However, we simply have to check the consecutive elements, and even if they are repeating like [1, 1, 2, 3, 4], they are not consecutive as they don't satisfy the condition nums[i]+1 == nums[i+1]. The primary focus is on the first statement, as the other one may confuse you to think of ambiguous corner cases. You can easily check for consecutive elements after sorting the array.
```

Comment:
1. add space 

e.g.

````
[x,x+n-1]
````

to

````
[x, x + n - 1]
````
2. use LaTex for variables.

e.g. 

````
$[1, 1, 2, 3, 4]$
````

````
nums[i] + 1 == nums[i + 1]
````

and so on